scanpipe/cyclonedx/__init__.py

```diff
@@ -0,0 +1,181 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# http://nexb.com and https://github.com/nexB/scancode.io
+# The ScanCode.io software is licensed under the Apache License version 2.0.
+# Data generated with ScanCode.io is provided as-is without warranties.
+# ScanCode is a trademark of nexB Inc.
+#
+# You may not use this software except in compliance with the License.
+# You may obtain a copy of the License at: http://apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+#
+# Data Generated with ScanCode.io is provided on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, either express or implied. No content created from
+# ScanCode.io should be considered or used as legal advice. Consult an Attorney
+# for any legal advice.
+#
+# ScanCode.io is a free software code scanning tool from nexB Inc. and others.
+# Visit https://github.com/nexB/scancode.io for support and download.
+
+import json
+import pathlib
+
+import jsonschema
+from hoppr_cyclonedx_models.cyclonedx_1_4 import Component
+from hoppr_cyclonedx_models.cyclonedx_1_4 import (
+    CyclonedxSoftwareBillOfMaterialsStandard as Bom_1_4,
+)
+
+CYCLONEDX_SPEC_VERSION = "1.4"
+CYCLONEDX_JSON_SCHEMA_LOCATION = "bom-1.4.schema.json"
+CYCLONEDX_JSON_SCHEMA_PATH = (
+    pathlib.Path(__file__).parent / CYCLONEDX_JSON_SCHEMA_LOCATION
+)
+CYCLONEDX_JSON_SCHEMA_URL = (
+    "https://raw.githubusercontent.com/"
+    "CycloneDX/specification/master/schema/bom-1.4.schema.json"
+)
+
+
+def get_bom(cyclonedx_document: dict):
+    """
+    Return CycloneDx BOM object.
+    """
+    return Bom_1_4(**cyclonedx_document)
+
+
+def get_components(bom: Bom_1_4):
+    """
+    Return list of components from CycloneDx BOM.
+    """
+    return recursive_component_collector(bom.components, [])
+
+
+def bom_attributes_to_dict(cyclonedx_attributes):
+    """
+    Return list dict from a list of CycloneDx attributes.
+    """
+    if not cyclonedx_attributes:
+        return {}
+
+    return [
+        json.loads(attribute.json(exclude_unset=True, by_alias=True))
+        for attribute in cyclonedx_attributes
+    ]
```

Comment:
What is a list dict in "Return list dict"? A list of dicts?
Also, we don't want code that may return a dict (line 62) or a list.
We should always return the same type.